lib/puppet/provider/user/windows_adsi.rb

```diff
@@ -123,8 +306,58 @@ def password
     user.password_is?( @resource[:password] ) ? @resource[:password] : nil
   end
 
+  # Sets the user's password. Takes the following options:
+  #
+  #   * force -- If true, the user's password will be set
+  #              no matter what.
+  #
+  def set_password(password, options = {})
+    force = options[:force] || false
+
+    # The force option is useful here because password_change_required will
+    # always be true after a user's created. Thus without the force option,
+    # we cannot set the user's password after creating them.
+    current_attributes = attributes
+    if ! force && (current_attributes[:password_change_required] == 'true')
```

Comment:
Should this warning message be printed and password not be reset? I kind of followed the model in https://github.com/caseywilliams/puppet/commit/5b1ec4339dc1e08c98385f1c9071ac7a0884584f#diff-cb6db5b98229b4ec070f9158795e3708R130 where we print warning messages and not reset the password for other situations.